server/controllers/allocationsController.ts

```diff
@@ -347,3 +370,105 @@ function toArrayNotation(href: string) {
 function fixupArrayNotation({ text, href }: { text: string; href: string }) {
   return { text, href: toArrayNotation(href) }
 }
+
+function getChoosePractitionerDataByTeam(
+  allocationInformationByTeam: ChoosePractitionerData,
+  teamNamesByCode: Map<string, string>
+): TeamOffenderManagersToAllocate[] {
+  const practitionerTeams = new Array<TeamOffenderManagersToAllocate>()
+  Object.entries(allocationInformationByTeam.teams).forEach(teamCodeAndPractitioner => {
+    const teamCode = teamCodeAndPractitioner[0]
+    const practitionersInTeam = (teamCodeAndPractitioner[1] as Practitioner[]).map(practitioner => {
+      const practitionerData = mapPractitioner(practitioner)
+      return {
+        ...practitionerData,
+        teamCode,
+        teamName: teamNamesByCode.get(teamCode),
+        selectionCode: TeamAndStaffCode.encode(teamCode, practitioner.code),
+      }
+    })
+    practitionersInTeam.sort(sortPractitionersByGrade)
+    practitionerTeams.push({
+      isPerTeam: true,
+      teamCode,
+      teamName: teamNamesByCode.get(teamCode),
+      offenderManagersToAllocate: practitionersInTeam,
+    })
+  })
+  return practitionerTeams
+}
+
+function getChoosePractitionerDataAllTeams(
```

Comment:
is there an opportunity to remove this function? could you call `getChoosePractitionerDataByTeam` first then loop through the result with a `reduce` combining the offenderManagersToAllocate into one list?